httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/HttpByteArrayCacheEntrySerializer.java

```diff
@@ -81,14 +115,65 @@ public class HttpByteArrayCacheEntrySerializer implements HttpCacheEntrySerializ
     private static final String SC_HEADER_NAME_NO_CONTENT = SC_CACHE_ENTRY_PREFIX + "no-content";
     private static final String SC_HEADER_NAME_VARIANT_MAP_KEY = SC_CACHE_ENTRY_PREFIX + "varmap-key";
     private static final String SC_HEADER_NAME_VARIANT_MAP_VALUE = SC_CACHE_ENTRY_PREFIX + "varmap-val";
-
     private static final String SC_CACHE_ENTRY_PRESERVE_PREFIX = SC_CACHE_ENTRY_PREFIX + "esc-";
 
-    private static final int BUFFER_SIZE = 8192;
 
+    /**
+     * The generator used to generate cached HTTP responses.
+     */
+    private final CachedHttpResponseGenerator cachedHttpResponseGenerator;
+
+    /**
+     * The size of the buffer used for reading/writing data.
+     */
+    private final int bufferSize;
+
+    /**
+     * Use the default, ASCII-only encoder for HTTP protocol and header values.
+     * It's the only thing that's widely used, and it's not worth it to support anything else.
+     */
+    private final SessionOutputBuffer outputBuffer;
```

Comment:
@arturobernalg Careful here. The class is expected to be stateless by the contract, but `SessionOutputBuffer` is not. It can not be an instance variable! You need to make it local.

Reply:
done.